Fix android transport mode hint

jiricrispeyn opened the pull request:
Updated `startTrip` bridge method on Android so it's consistent with iOS. 

- Passing an integer from JavaScript will now correctly convert to the according TransportMode value
- Convert metadata to HashMap so it can be used inside a startTrip method
- Add missing parameters to startTrip method

sebouh00 commented:
For TransportMode hint, this will work granted that PR #43 is merged and the readme maps the numbers correctly. But I'm not a big fan of this approach. It relies on the inherent values set to the enum by the compile. If for some reason the other of the enums change on the SDK, the mapping will be incorrect. A better approach is to explicitly map numbers RN uses to enum values the SDK uses.

So mapping in readme:

In the RN module:

```@Nullable
    private TransportMode toTransportMode(int t) {
        switch (t) {
            case 1:
                return CAR;
            case 2:
                return BICYCLE;
            case 3:
                return ON_FOOT;
            case 4:
                return TRAIN;
            case 5:
                return TRAM;
            case 6:
                return BUS;
            case 7:
                return PLANE;
            case 8:
                return BOAT;
            case 9:
                return METRO;
            case 10:
                return RUNNING;
            default:
                return null;
        }
    }

jiricrispeyn commented:
@sebouh00, I've corrected the incorrect import as mentioned in PR #43.
I also added the `toTransportMode` method you've proposed.

If this looks good to you I'll release this later today.
